### PR TITLE
Use external hal

### PR DESCRIPTION
**Pull request by whikloj**
**GitHub Issue**: #63 

# What does this Pull Request do?

Includes contrib module HAL dependency

# What's new?
* Does this change require documentation to be updated? no
* Does this change add any new dependencies? yes
* Does this change require any other modifications to be made to the repository
(i.e. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# How should this be tested?

Still be able to produce JSON-LD serializations of Drupal objects.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/8-x-committers

### CODE REVIEW

**Comment by whikloj** [comment]
This PR will not be merged until Drupal 10 is released.

**Comment by whikloj** [comment]
This replaces #65

**Comment by rosiel** [comment]
suggestion to make this a 3.x branch of jsonld (because changing the supported Drupal from 9 only to 10 only would make this a major version, wouldn't it?)

**Comment by whikloj** [comment]
That is fine by me, probably makes sense not to rush to create a `3.x` branch yet as any other PRs would then need to be applied to both `2.x` and `3.x`. 

IMHO Better to wait until Drupal 10 is actually released and we are starting to make sure we (Islandora) are compliant.

**Comment by kayakr** [comment]
@whikloj @rosiel It would be good to have `core_version_requirement: ^9 || ^10` so that this module can applied to D9 sites as they prepare for Drupal 10.

**Comment by rosiel** [comment]
Yeah, D10's been released so we should get on this! 

From what I gather, the composer requirement for the external hal is needed if and only if we're on Drupal 10, but it will conflict with Drupal 9.  I don't think it's possible (please correct me if i'm wrong) for there to be a branch where the Composer file is compatible with 9 or 10. 

I think (again, I hope y'all know more than me) that we'll need to create a jsonld:3.x branch which will be for D10 and only D10. This PR (with a few modifications) can start this off. But islandora/islandora currently has a requirement for `jsonld:^2`. I think if we change that to >2, then we'll be able to start moving forward in testing with Drupal 10. There will be other problems, guaranteed, but it'll be a start.

File: .github/workflows/build-2.x.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        php-versions: ["7.4", "8.0", "8.1"]
-        drupal-version: ["9.3.x", "9.4.x-dev"]
+        php-versions: ["8.0", "8.1"]
+        drupal-version: ["10.0.x-dev", "10.1.x-dev"]
```
Comment (rosiel) [inline .github/workflows/build-2.x.yml:25]: "10.0" instead of "10.0.x-dev"

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
         }
     ],
     "require" : {
-        "php": ">=7.4"
+        "php": ">=8.0",
+        "drupal/hal-hal": "^1.0"
```
Comment (rosiel) [inline composer.json:24]: For D10, this seems it should be `"drupal/hal": "^2"` at least per https://www.drupal.org/project/hal

File: jsonld.info.yml [hunk 3]
```diff
@@ -2,8 +2,8 @@ name: 'JSON-LD'
 type: module
 description: 'Serializes entities to JSON-LD / LDP for the Islandora Project'
 package: Web services
-core_version_requirement: ^9
+core_version_requirement: ^10
 dependencies:
-  - hal
-  - serialization
-  - rdf
+  - drupal:hal
```
Comment (rosiel) [inline jsonld.info.yml:7]: should be `hal:hal` etc.

**Comment by whikloj** [comment]
Closing in favour of #69

**Pull request closed without merging**